src/Microsoft.AspNetCore.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -55,19 +55,19 @@ public DateHeaderValueManager()
         /// Returns a value representing the current server date/time for use in the HTTP "Date" response header
         /// in accordance with http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18
         /// </summary>
-        /// <returns>The value.</returns>
-        public virtual string GetDateHeaderValue()
+        /// <returns>The value in string and byte[] format.</returns>
+        public DateHeaderValues GetDateHeaderValues()
         {
-            _hadRequestsSinceLastTimerTick = true;
-            PrepareDateValues();
-            return _dateValue;
-        }
+            if (!_hadRequestsSinceLastTimerTick)
+            {
+                PrepareDateValues();
+            }
 
-        public byte[] GetDateHeaderValueBytes()
-        {
-            _hadRequestsSinceLastTimerTick = true;
-            PrepareDateValues();
-            return _activeDateBytes ? _dateBytes0 : _dateBytes1;
+            return new DateHeaderValues()
+            {
+                Bytes = _activeDateBytes ? _dateBytes0 : _dateBytes1,
```

Comment:
Reading is very hot; updating is once per second. As writing is very slow; probably not saving much from not allocating as its only a short array - though at high throughput each one is likely to hit a higher GC generation. 

Might be safest to allocate on each write? As some requests may take seconds or minutes depending on db access etc.